lib/resty/jwt.lua

```diff
@@ -466,9 +496,11 @@ function _M.sign(self, secret_key, jwt_obj)
   local alg = jwt_obj[str_const.header][str_const.alg]
   local signature = ""
   if alg == str_const.HS256 then
-    signature = hmac:new(secret_key, hmac.ALGOS.SHA256):final(message)
+    local secret_str = get_secret_str(secret_key, jwt_obj)
```

Comment:
Yes - it does, but I was following the pattern for other errors in the header (such as on line 484 where `typ` is invalid and just raises an error directly).

I can use a `pcall` if you prefer, though.